pkg/usage/metrics.go

```diff
@@ -0,0 +1,76 @@
+package usage
+
+import (
+	"fmt"
+	"time"
+)
+
+var (
+	NoMetricsNameErr           = fmt.Errorf("metrics' name must be specified")
+	NoMetricsTimeRangeErr      = fmt.Errorf("metrics' time range (from/to - ISO8601 format) must be specified")
+	InvalidMetricsTimeRangeErr = fmt.Errorf("invalid time range for metrics")
+)
+
+// MetricsRequest represents a client request for metrics based on time range
+type MetricsRequest struct {
+	Name string    `json:"name"`
+	From time.Time `json:"from"`
+	To   time.Time `json:"to"`
+}
+
+func (mr MetricsRequest) Valid() (bool, error) {
```

Comment:
Nit: the boolean isn't needed because the existence of an error already tells you whether it's valid

Reply:
updated